src/main/scala/scorex/core/NodeViewHolder.scala

```diff
@@ -1,393 +1,306 @@
 
-  case class UpdateInformation(history: HIS,
-                               state: MS,
-                               failedMod: Option[PMOD],
-                               alternativeProgressInfo: Option[ProgressInfo[PMOD]],
-                               suffix: IndexedSeq[PMOD])
+  /** Underlying memory pool actor
+    */
+  protected val memoryPoolActor: ActorRef = createMemoryPoolActor()
+
+  protected var vault: Vault = restoreVault().getOrElse(genesisVault())
 
   val scorexSettings: ScorexSettings
 
-  /**
-    * Cache for modifiers. If modifiers are coming out-of-order, they are to be stored in this cache.
-    */
-  protected lazy val modifiersCache: ModifiersCache[PMOD, HIS] =
-    new DefaultModifiersCache[PMOD, HIS](scorexSettings.network.maxModifiersCacheSize)
-
-  /**
-    * The main data structure a node software is taking care about, a node view consists
-    * of four elements to be updated atomically: history (log of persistent modifiers),
-    * state (result of log's modifiers application to pre-historical(genesis) state,
-    * user-specific information stored in vault (it could be e.g. a wallet), and a memory pool.
-    */
-  private var nodeView: NodeView = restoreState().getOrElse(genesisState)
+  protected implicit val defaultAskTimeout: Timeout = Timeout(10.seconds)
+  protected implicit val executionContext: ExecutionContext = context.dispatcher
 
-  /**
-    * Restore a local view during a node startup. If no any stored view found
+  /** Restore a local history during a node startup. If no any stored view found
     * (e.g. if it is a first launch of a node) None is to be returned
     */
-  def restoreState(): Option[NodeView]
+  protected def restoreHistory(): Option[History]
 
-  /**
-    * Hard-coded initial view all the honest nodes in a network are making progress from.
+  /** Restore a local state during a node startup. If no any stored view found
+    * (e.g. if it is a first launch of a node) None is to be returned
     */
-  protected def genesisState: NodeView
+  protected def restoreState(): Option[State]
 
+  /** Restore a local vault during a node startup. If no any stored view found
+    * (e.g. if it is a first launch of a node) None is to be returned
+    */
+  protected def restoreVault(): Option[Vault]
 
-  protected def history(): HIS = nodeView._1
-
-  protected def minimalState(): MS = nodeView._2
+  /** Restore a local memory pool during a node startup. If no any stored view found
+    * (e.g. if it is a first launch of a node) None is to be returned
+    */
+  protected def restoreMempool(): Option[MPool]
 
-  protected def vault(): VL = nodeView._3
+  /** Hard-coded initial history all the honest nodes in a network are making progress from
+    */
+  protected def genesisHistory(): History
 
-  protected def memoryPool(): MP = nodeView._4
+  /** Hard-coded initial state all the honest nodes in a network are making progress from
+    */
+  protected def genesisState(): State
```

Comment:
How these direct readers are expected to be used from NVH when state and history aren't available directly from NVH?

Reply:
This was expected to be used for component actors initialization, as in Hybrid example. But you're probably right, this should be moved to actors itself